plugin-server/src/main/services/http-server.ts

```diff
@@ -1,187 +1,177 @@
 import * as prometheus from 'prom-client'
 
 import { status } from '../../utils/status'
+import { delay } from '../../utils/utils'
 
 prometheus.collectDefaultMetrics()
 const v8Profiler = require('v8-profiler-next')
 v8Profiler.setGenerateType(1)
 
-export function createHttpServer(
-    port: number,
+export const expressApp: express.Application = express()
+
+export function setupCommonRoutes(
     healthChecks: { [service: string]: () => Promise<boolean> | boolean },
     analyticsEventsIngestionConsumer?: KafkaJSIngestionConsumer | IngestionConsumer
-): Server {
-    const server = createServer(async (req: IncomingMessage, res: ServerResponse) => {
-        if (req.method !== 'GET') {
-            res.statusCode = 404
-            res.end()
-            return
-        }
-        if (req.url === '/_health') {
-            // Check that all health checks pass. Note that a failure of these
-            // _may_ result in the process being terminated by e.g. Kubernetes
-            // so the stakes are high.
-            //
-            // Also, Kubernetes will call this endpoint frequently, on each pod,
-            // so we want to make sure it's fast and doesn't put any stress on
-            // other services. Ideally it shouldn't make any calls to other
-            // services.
-            //
-            // Here we take all of the health checks we are given, run them in
-            // parallel, and return the results. If any of the checks fail, we
-            // return a 503 status code, otherwise we return a 200 status code.
-            //
-            // In all cases we should return a JSON object with the following
-            // structure:
-            //
-            // {
-            //   "status": "ok" | "error",
-            //   "checks": {
-            //     "service1": "ok" | "error",
-            //     "service2": "ok" | "error",
-            //     ...
-            //   }
-            // }
-            const checkResults = await Promise.all(
-                // Note that we do not use `Promise.allSettled` here so we can
-                // assume that all promises have resolved. If there was a
-                // rejected promise, the http server should catch it and return
-                // a 500 status code.
-                Object.entries(healthChecks).map(async ([service, check]) => {
-                    try {
-                        return { service, status: (await check()) ? 'ok' : 'error' }
-                    } catch (error) {
-                        return { service, status: 'error', error: error.message }
-                    }
-                })
-            )
-
-            const statusCode = checkResults.every((result) => result.status === 'ok') ? 200 : 503
+): express.Application {
+    expressApp.get('/_health', getHealth(healthChecks))
+    expressApp.get('/_ready', getReady(analyticsEventsIngestionConsumer))
```

Comment:
In my head types and tests are the saving grace here more than namings but I see your point!

Reply:
Oh, fair enough! I don't know express. In our case we got bit because a `String` response from our metrics fn was totally fine for the HTTP service routing layer we used.